forge-gui/res/cardsfolder/upcoming/vexing_puzzlebox.txt

```diff
@@ -0,0 +1,11 @@
+Name:Vexing Puzzlebox
+ManaCost:3
+Types:Artifact
+T:Mode$ RolledDieOnce | TriggerZones$ Battlefield | ValidPlayer$ You | Execute$ TrigPutCounter | TriggerDescription$ Whenever you roll one or more dice, put a number of charge counters on CARDNAME equal to the result.
```

Comment:
This needs some engine support to work correctly, because currently it only counts naturally like on _Netherese Puzzle-Ward_:
> Some effects may modify the result of a die roll. This may be part of the instruction to roll a die or it may come from other cards. Anything that references the “result” of a die roll is looking for the result after these modifications. Anything that is looking for the “natural result” is looking for the number shown on the face of the die before these modifications.

Reply:
There are a lot of rolleddieonce I found, does it affects all rooledieonce trigger? if it is then we should remove all rolleddieonce cards

Reply:
Well most cards like `Brazen Dwarf` don't really care about the result so this is more of a corner case for now

Feel free to turn this into an issue for me, the card should work for now

Reply:
- Issue #1154